### PR TITLE
setup rpc tests and spell infra

### PR DESCRIPTION
**Pull request by ilinzweilin**
- added basic setup for rpc tests from current chain forks
- added first rpc test to check if deployed contracts are wired correctly
- added spell template + spell test (using rpc test setup as base)
-  added spell to migrate first LP contracts

### CODE REVIEW

File: test/spells/Spell.sol [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    function asset() external returns (address);
+}
+
+// Spell to mirate the LiquidityPool factory contract
+contract Spell is Addresses {
+    bool public done;
+    string public constant description = "Liquidity Pool Factory migration spell";
+
+    address public constant LIQUIDITY_POOL_FACTORY_NEW = address(0x8273E36EEcf7A8604BEdEe68FC24Af121B64f165);
+    address public constant DEPRICATED_LIQUIDITY_POOL = address(0xa0872E8D2975483b2Ab4Afcee729133D8666F6f5);
```
Comment (hieronx) [inline test/spells/Spell.sol:36]:
```suggestion
    address public constant DEPRECATED_LIQUIDITY_POOL = address(0xa0872E8D2975483b2Ab4Afcee729133D8666F6f5);
```

File: test/rpc/RPC.t.sol [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+import {Root} from "src/Root.sol";
+import {InvestmentManager} from "src/InvestmentManager.sol";
+import {PoolManager} from "src/PoolManager.sol";
+import {Escrow} from "src/Escrow.sol";
+import {UserEscrow} from "src/UserEscrow.sol";
+import {TrancheToken} from "src/token/Tranche.sol";
+import {Gateway} from "src/gateway/Gateway.sol";
+import {AxelarRouter} from "src/gateway/routers/axelar/Router.sol";
+import {TrancheTokenFactory, LiquidityPoolFactory, RestrictionManagerFactory} from "src/util/Factory.sol";
+import {DelayedAdmin} from "src/admins/delayedAdmin.sol";
```
Comment (hieronx) [inline test/rpc/RPC.t.sol:15]:
```suggestion
import {DelayedAdmin} from "src/admins/DelayedAdmin.sol";
```

File: test/rpc/RPC.t.sol [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+import {PauseAdmin} from "src/admins/pauseAdmin.sol";
```
Comment (hieronx) [inline test/rpc/RPC.t.sol:16]:
```suggestion
import {PauseAdmin} from "src/admins/PauseAdmin.sol";
```

File: test/spells/Spell.sol [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        migrateLiquidtyPoolFactory();
+        removeLiquidityPool();
+
+        // revoke all permissions from spell
+        AuthLike(address(root)).deny(self);
+        AuthLike(LIQUIDITY_POOL_FACTORY_NEW).deny(self);
+        AuthLike(poolManager).deny(self);
+    }
+
+    function migrateLiquidtyPoolFactory() internal {
```
Comment (hieronx) [inline test/spells/Spell.sol:64]:
```suggestion
    function migrateLiquidityPoolFactory() internal {
```

File: test/spells/Spell.sol [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    function getLiquidityPool(uint64 poolId, bytes16 trancheId, address currency) external returns (address);
+}
+
+interface LiquidityPoolLike {
+    function poolId() external returns (uint64);
+    function trancheId() external returns (bytes16);
+    function asset() external returns (address);
+}
+
+// Spell to mirate the LiquidityPool factory contract
```
Comment (AStox) [inline test/spells/Spell.sol:30]:
```suggestion
// Spell to migrate the LiquidityPool factory contract
```

File: test/spells/Spell.t.sol [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        // admin submits a tx to delayedAdmin in order to rely spell -> to be done manually before spell cast
+        vm.prank(admin);
+        DelayedAdmin(delayedAdmin).scheduleRely(address(spell));
+        // warp to the time when the spell can be cast -> current block + delay
+        vm.warp(block.timestamp + Root(root).delay());
+        Root(root).executeScheduledRely(address(spell)); // --> to be called after delay has passed
+        spell.cast();
+    }
+
+    function testCastSuccessfull() public {
```
Comment (AStox) [inline test/spells/Spell.t.sol:35]:
```suggestion
    function testCastSuccessful() public {
```

**Review by AStox — COMMENTED** [review]
Looks great, but I'd like to suggest using the term Fork Tests instead of RPC tests everywhere to better align with foundry's naming. I think we got "RPC tests" from dapp-tools, right? What do you guys think? @hieronx @ilinzweilin

**Review by hieronx — APPROVED** [review]

**Pull request merged**